src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -621,18 +712,28 @@ private List<string> GatherInternalParams(Function function, out CSharpTypePrint
 
                 var typeName = param.CSharpType(TypePrinter);
 
-                @params.Add(string.Format("{0} {1}", typeName, param.Name));
+                @params.Add(string.Format("{0} {1}", typeName, useOriginalParamNames ? param.Name : "param" + i++));
+                if (allTypesInOne != null)
+                    allTypesInOne.Result.Add(Regex.Replace(typeName.ToString(), pattern, "_"));
 
                 if (param.Kind == ParameterKind.IndirectReturnType &&
                     isInstanceMethod && Options.IsItaniumLikeAbi)
+                {
                     @params.Add("global::System.IntPtr instance");
+                    if (allTypesInOne != null)
+                        allTypesInOne.Result.Add("global_System_IntPtr");
+                }
             }
 
             if (method != null && method.IsConstructor)
             {
                 var @class = (Class) method.Namespace;
                 if (Options.IsMicrosoftAbi && @class.Layout.HasVirtualBases)
+                {
                     @params.Add("int " + GeneratedIdentifier("forBases"));
+                    if (allTypesInOne != null)
+                        allTypesInOne.Result.Add("int_forBases");
```

Comment:
"forBases" is not part of the type but rather an identifier so I think just "int" would be enough.